src/main/scala/vsys/settings/WebhookEventRules.scala

```diff
@@ -0,0 +1,164 @@
+package vsys.settings
+
+import com.typesafe.config.Config
+import scala.util.Try
+import net.ceedubs.ficus.Ficus._
+import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+import net.ceedubs.ficus.readers.ValueReader
+
+trait WebhookEventRules extends Product with Serializable {
+  //override def applyRule(valIn: Any*): Boolean
+  //TO DO
+  val value: Any 
+}
+
+trait EventConfigReader {
+  val field: String
+  private val path = "rules"
+  lazy val valPath = path + "." + field
+
+  def fromConfig(config: Config): Option[WebhookEventRules]
+
+  def read[T: ValueReader](config: Config): Option[T] = {
+    Try(config.as[T](valPath)).toOption
+  }
+}
+
+
+case class AfterHeight(value: Long) extends WebhookEventRules
+case class AfterTime(value: Long) extends WebhookEventRules
+case class WithTxs(value: Boolean) extends WebhookEventRules
+case class WithMintingTxs(value: Boolean) extends WebhookEventRules
+case class RelatedAccs(value: Seq[String]) extends WebhookEventRules
+case class IncludeTypes(value: Seq[Int]) extends WebhookEventRules
+case class ExcludeTypes(value: Seq[Int]) extends WebhookEventRules
+case class AmtGT(value: Long) extends WebhookEventRules
+case class AmtLT(value: Long) extends WebhookEventRules
+case class AmtGTE(value: Long) extends WebhookEventRules
+case class AmtLTE(value: Long) extends WebhookEventRules
+case class AmtWithFee(value: Boolean) extends WebhookEventRules
+case class WithTxsOfTypes(value: Seq[Int]) extends WebhookEventRules
+case class WithTxsOfAccs(value: Seq[String]) extends WebhookEventRules
+case class WithStateOfAccs(value: Seq[String]) extends WebhookEventRules
+
+
+
+object AfterHeight extends EventConfigReader {
```

Comment:
companion object should be right below case class

Reply:
sure~~